docpages/example_programs/the_basics/editing-objects.md

```diff
@@ -0,0 +1,92 @@
+\page editing-objects Editing Objects (like messages)
+
+Sometimes we need to update an object, such as message or channel. At first, it might seem confusing, but it's actually really simple! It's done with [callbacks](https://dpp.dev/callback-functions.html). You just need to use an object with identical properties you don't need to update.
+
+~~~~~~~~~~{.cpp}
+#include <dpp/dpp.h>
+
+int main() {
+    dpp::cluster bot("Token is right here", dpp::i_default_intents | dpp::i_message_content);
+    /* the second argument is a bitmask of intents - i_message_content is needed to get messages */
+
+    bot.on_log(dpp::utility::cout_logger());
+
+    /* The event is fired when someone issues your commands */
+    bot.on_slashcommand([&bot](const dpp::slashcommand_t& event) -> void {
+        if (event.command.get_command_name() == "msg-send") {
+            event.reply("That's a message");
+        } else if (event.command.get_command_name() == "msg-edit") {
+            std::string content = std::get<std::string>(event.get_parameter("content"));
+
+            /* get message to edit it after */
+            dpp::snowflake msg_id = std::get<std::string>(event.get_parameter("msg-id"));
+            /* here string will automatically be converted to snowflake */
+
+            bot.message_get(msg_id, event.command.channel_id, [&bot, content, event](const dpp::confirmation_callback_t& callback) -> void {
+                if (callback.is_error()) {
+                    event.reply("error");
+                    return;
+                }
+                auto message = callback.get<dpp::message>();
+
+                /* change the message content and edit the message itself */
+                message.set_content(content);
+                bot.message_edit(message);
+                event.reply("Message content is now `" + content + "`.");
+            });
+        } else if (event.command.get_command_name() == "channel-edit") {
+            auto name = std::get<std::string>(event.get_parameter("name"));
```

Comment:
You've used `auto` here but not on line 19? This is inconsistent, I'd say to change this to `std::string name = std::get<...>(...)` or change line 19 to auto.

Reply:
Not sure how I missed it but on my way

Reply:
would also put const whenever possible so beginners will be more likely to inherit the habit of using const